### DISCUSSION TITLE
Using OpenTelemetry as collector for ForgeRock with context propagation.

### DISCUSSION THREAD

**Discussion by deepakchanalia** [open-telemetry/opentelemetry-collector-contrib#7941] [Q&A] [answered]
Hello,
We need to use opentelemetry as a collector/exporter of third party (e.g. Forgerock IDAM suite) product logs to Splunk or NewRelic dashboard for event monitoring and log analysis. Below is the high-level flow of log data:

**FORGEROCK (AM,IG,DJ,IDM) -> OPENTELEMETRY (COLLECTOR/AGENT ETC.) -> SPLUNK (OR NEW RELIC).**

With this setup/configuration in place, we want to implement the context propagation feature as well (provided by OpenTelemetry) in which we should be able to track the requests in different services under same parent-id or trace-id. 
Considering the ForgeRock products (i.e. OpenAM, OpenDJ, Open IG etc) are third party products and not using the opentelemetry API/SDKs to handle the context propagation headers that OpenTelemetry recognizes. Could you please suggest if it is possible to collect the default logs (aligned with context propagation headers provided by opentelemetry) of a product (not instrumented with OpenTelemetry APIs/SDK) using the opentelemetry agent/collector (like we are doing) and send them to other third party analyzer tool (Splunk/NewRelic etc) ? 
If this configuration is possible, could you please point us some example/sample which we can refer in order to implement it.
Thanks
Deepak

**Comment by devrimdemiroz** [comment, marked as answer]
@deepakchanalia , I had similar stuff by the end of 2020. My notes(otel/loki/prometheus/jaeger) are here -> https://devrimdemiroz.gitlab.io/post/2021-03-09-observeopenam/ 

Hope it helps. You can collect and export logs and metrics in various ways. However, without opentelemetry tracing **auto**-instrumentation, it is very hard to analyze by trace-id mentioned in Forgerock documents.

**Reply by deepakchanalia** [reply]
Hello @devrimdemiroz Thanks a lot for your response on this thread. Actually I did something similar and was able to achieve the intended outcome. Thanks
